projects/plugins/jetpack/extensions/blocks/map/controls.js

```diff
@@ -0,0 +1,215 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import {
+	Button,
+	ButtonGroup,
+	ExternalLink,
+	PanelBody,
+	TextControl,
+	ToggleControl,
+	ToolbarButton,
+	ToolbarGroup,
+	RangeControl,
+	BaseControl,
+} from '@wordpress/components';
+
+import {
+	BlockAlignmentToolbar,
+	PanelColorSettings,
+} from '@wordpress/block-editor';
+
+/**
+ * Internal dependencies
+ */
+import Locations from './locations';
+import { settings } from './settings.js';
+
+export default ( {
+    attributes,
+    setAttributes,
+    state,
+    setState,
+    context,
+    mapRef,
+    instanceId,
+    minHeight,
+    removeAPIKey,
+    updateAPIKey,
+    setPointVisibility,
+} ) => {
+    const updateAlignment = value => {
+		setAttributes( { align: value } );
+
+		// Allow one cycle for alignment change to take effect
+		setTimeout( mapRef.current.sizeMap, 0 );
+	};
+
+    /**
+     * Change event handler for the map height sidebar control. Ensures the height is valid,
+     * and updates both the height attribute, and the map component's height in the DOM.
+     *
+     * @param {Event} event - The change event object.
+     */
+    const onHeightChange = event => {
+		const { mapHeight } = attributes;
+
+		let height = parseInt( event.target.value, 10 );
+
+		if ( isNaN( height ) ) {
+			// Set map height to default size and input box to empty string
+			height = null;
+		} else if ( null == mapHeight ) {
+			// There was previously no height defined, so set the default.
+			height = mapRef.current.mapRef.current.offsetHeight;
+		} else if ( height < minHeight ) {
+			// Set map height to minimum size
+			height = minHeight;
+		}
+
+		setAttributes( {
+			mapHeight: height,
+		} );
+
+		setTimeout( mapRef.current.sizeMap, 0 );
+	};
+
+    if ( context === 'toolbar' ) {
```

Comment:
Oh, neat way of handling both sets of controls from the one component 👍, if it was much more complex, I'd say we could split this out into two separate components, but this looks nice and simple to keep here.

Reply:
Sounds good. I kept it this way since it was the same as the whatsapp tests. :+1: